internal/context/context.go

```diff
@@ -27,20 +28,72 @@ type OctantClientState string
 
 const ClientStateKey = OctantClientState("clientState")
 
-type ClientState struct {
+func NewClientState(cid string, f []octant.Filter, n string, cn string) clientState {
+	return clientState{
+		clientID:    cid,
+		filters:     f,
+		namespace:   n,
+		contextName: cn,
+	}
+}
+
+func ClientStateFromJson(jsonBytes []byte) (clientState, error) {
+	var tmpState jsonClientState
+	if err := json.Unmarshal(jsonBytes, &tmpState); err != nil {
+		return clientState{}, err
+	}
+
+	return NewClientState(tmpState.ClientID, tmpState.Filters, tmpState.Namespace, tmpState.ContextName), nil
+}
+
+type jsonClientState struct {
 	ClientID    string          `json:"clientID"`
 	Filters     []octant.Filter `json:"filters"`
 	Namespace   string          `json:"namespace"`
 	ContextName string          `json:"contextName"`
 }
 
-func WithClientState(ctx context.Context, state ClientState) context.Context {
+type clientState struct {
```

Comment:
Just out of curiosity, why is the `client` in `clientCase` lowercase here when it was previously uppercase

Reply:
This is a go convention to make something private, this helps us enforce folks using the interference and helper methods instead of the struct directly

Reply:
Got it, thanks!